neo/test/coretest/test_analogsignal.py

```diff
@@ -466,6 +466,31 @@ def test__time_slice_should_set_parents_to_None(self):
         self.assertEqual(result.segment, None)
         self.assertEqual(result.channel_index, None)
 
+    def test__time_slice_close_to_sample_boundaries(self):
+        # see issue 530
+
+        sig = AnalogSignal(np.arange(25000) * pq.uV,
+                           t_start=0 * pq.ms,
+                           sampling_rate=25 * pq.kHz)
+
+        window_size = 3.0 * pq.ms
+
+        expected_shape = int(np.rint((window_size * sig.sampling_rate).simplified.magnitude))
+
+        # test with random times
+        t_start = (window_size / 2).magnitude
+        t_stop = (sig.t_stop.rescale(pq.ms) - window_size / 2).magnitude
+        for t in np.random.uniform(t_start, t_stop, size=1000):
```

Comment:
Maybe we could put a seed here or even better voltontary loop over a ambiguous float at sub sample resolution.

Reply:
I think random values without a seed are better in this case, because we'll get different values every time the test is run, and hence over time the tests will cover a much wider range of values.

Reply:
I think a seed makes more sense, because with a true random component in the tests (and potentially randomly failing tests) unittests get quite hard to debug.

Reply:
I agree with using a seed in cases where there is a small but finite chance of the test failing for certain random values. In this case, however, if the code is correct then the probability of failure  should be zero.

Reply:
Ok, I see your point. Let's use the current implementation for now.

Reply:
I am OK for no seed.